### PR TITLE
gib hitchhikers on ftl

### LINKED ISSUE #18441: escape on emergency shuttle should require being inside
## Description
Unwrenching a thruster on the emergency shuttle and sitting on the lattice shouldn't count as being on the shuttle. People are also RCDing floors down to escape on the outside

### PR DESCRIPTION
**Pull request by deltanedas**
## About the PR
if you are on lattice or space when ftl starts or ends you will get gibbed

fixes #18441

## Why / Balance
see issue

## Technical details
checks tile under every mob to see if it should get gibbed

caveat of it is it does not check recursively so you can survive in a crate/locker/disposal unit/a dragons stomach/ripley

## Media

A) skeletony gibbed when ftl starts, skelebro is ok since he is on plating
B) skelebro is pushed onto lattice when ftl ends, he is gibbed

https://github.com/space-wizards/space-station-14/assets/39013340/de6679e1-430d-4723-856b-61fed529a6b4

ok sound is sus it seems to be played only for the person being gibbed, but using bang stick on someone else plays sound still... very sus

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
no

**Changelog**
:cl:
- tweak: A fracture in bluespace is now causing spacemen that hitchhike on shuttles to be gibbed before and after FTL travel. Keep legs inside the vehicle at all times.

### CODE REVIEW

**Comment by LankLTE** [comment]
This could also lead to people being gibbed after shuttle bombing, which isn't ideal

**Comment by deltanedas** [comment]
1. having to define what lattice is inside the shuttle as opposed to inside seems like hell, cant check atmos since its spaced
2. they kinda would fly out the top/bottom...
3. shuttle bombings cant be repaired so they are typically cordoned off by pros, reducing chance of someone standing on a lattice inside before getting to centcom

**Comment by Equivocateur** [comment]
> This could also lead to people being gibbed after shuttle bombing, which isn't ideal

https://youtu.be/S6v252cYhp0?si=Zw2FdfFPnapN2AHg&t=90 but in reverse, makes sense in or outside the shuttle during ftl either way.

**Comment by deltanedas** [comment]
real

**Comment by CaptainSqrBeard** [comment]
People will just bring a steel sheet to place on lattice. That's probably not the solution

**Comment by deltanedas** [comment]
you cant do that

**Comment by CaptainSqrBeard** [comment]
> you cant do that

what stops me from doing that?

**Comment by coolmankid12345** [comment]
It literally doesn't work

**Comment by HoidC** [comment]
ok hear me out but there’s two solutions to this 

1. Zones (I’m sent to hell for uttering this word)
2. Making shuttle tiles bomb proof so people don’t get instagibbed by minibombs, also reduces the ridiculous power of the end round shuttle bomb

**Comment by deltanedas** [comment]
for whatever reason adding tiles to lattice is counted as building for evac

**Comment by UbaserB** [comment]
zones sound fine honestly. Atmos markers already exist, so id image it’d be similar.

**Comment by CaptainSqrBeard** [comment]
I didn't know that you can't build tiles on shuttle. But as it doesn't check containers I can build a crate and hide inside 🤔

**Comment by deltanedas** [comment]
yes its unfortunate but not sure how i would fix it

**Comment by JoeHammad1844** [comment]
thing is that  escape pods dont count for the escape to centcom alive syndie objective, so if you go loud you just wont be able to complete th objective because sec will catch you

File: Content.Server/Shuttles/Systems/ShuttleSystem.FasterThanLight.cs [hunk 1]
```diff
@@ -489,6 +500,31 @@ private void KnockOverKids(TransformComponent xform, EntityQuery<BuckleComponent
         }
     }
 
+    private void GibKids(TransformComponent xform, ref ValueList<(EntityUid, BodyComponent)> toGib)
+    {
+        var bodyQuery = GetEntityQuery<BodyComponent>();
+        var xformQuery = GetEntityQuery<TransformComponent>();
```
Comment (metalgearsloth) [inline Content.Server/Shuttles/Systems/ShuttleSystem.FasterThanLight.cs:506]: Chuck it on the system.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

**Comment by deltanedas** [comment]
> so if you go loud you just wont be able to complete th objective because sec will catch you

hiding on thruster is cowardice
if you want to go load bomb the shittle and be the only one left

**Review by metalgearsloth — APPROVED** [review]

**Comment by JoeHammad1844** [comment]
> hiding on thruster is cowardice if you want to go loud bomb the shittle and be the only one left

no TC left :sad:

**Pull request merged**